tests/unit/via/services/checkmate_test.py

```diff
@@ -0,0 +1,133 @@
+from unittest.mock import create_autospec, sentinel
+
+import pytest
+from checkmatelib import BadURL as CheckmateBadURL
+from checkmatelib import CheckmateException
+from checkmatelib.client import BlockResponse
+from pyramid.httpexceptions import HTTPTemporaryRedirect
+
+from via.exceptions import BadURL
+from via.services.checkmate import CheckmateService, factory
+
+
+class TestCheckmateService:
+    def test_check_url(self, checkmate_client, svc):
+        result = svc.check_url(sentinel.url)
+
+        checkmate_client.check_url.assert_called_once_with(
+            sentinel.url,
+            allow_all=sentinel.allow_all,
+            blocked_for=sentinel.blocked_for,
+            ignore_reasons=sentinel.ignore_reasons,
+        )
```

Comment:
By using composition instead of inheritance these assertions are much clearer. You can clearly see that it's asserting that a method on `checkmate_client` was called. And if you're not sure what `checkmate_client` is you can look down at that fixture and see exactly what it is, but the clear and consistent naming should make it pretty obvious: it's the `checkmate_client` argument to `CheckmateService.__init__()`, saved as `self._checkmate_client` within `CheckmateService`, and it's an instance of `checkmatelib`'s `CheckmateClient` class.

Because the old `ViaCheckmateClient` class **inherited** from `checkmatelib.CheckmateClient` rather than composing it had to do something much more confusing here:

For example there was this test that has no assertions and that writes some private attributes of `checkmatelib` (an external library's) `CheckmateClient`:

https://github.com/hypothesis/via/blob/d19c83df23de7b4d67b0d8aba37d218b3b74ade0/tests/unit/via/checkmate_test.py#L14-L17

The tests also had a fixture that patches one of the methods of the `ViaCheckmateClient` object-under-test, and even more confusingly the method being patched is a method that's not actually implemented in `ViaCheckmateClient` itself, it's inherited from `checkmatelib.CheckmateClient`, so what this is actually doing is reaching into `checkmatelib` to patch one of its methods:

https://github.com/hypothesis/via/blob/d19c83df23de7b4d67b0d8aba37d218b3b74ade0/tests/unit/via/checkmate_test.py#L64-L68

The tests then had assertions like `check_url.assert_called_once_with()` which are confusing because there is no `check_url()` method defined anywhere in Via (it's a method that Via implicitly _inherits_ from `checkmatelib`):

https://github.com/hypothesis/via/blob/d19c83df23de7b4d67b0d8aba37d218b3b74ade0/tests/unit/via/checkmate_test.py#L28-L33